src/aics-image-viewer/shared/utils/hooks.ts

```diff
@@ -0,0 +1,25 @@
+import { useState, useRef, useCallback } from "react";
+
+/** A `useState` that also creates a getter function for breaking through closures. */
+export function useStateWithGetter<T>(initialState: T | (() => T)): [T, (value: T) => void, () => T] {
+  const [state, setState] = useState(initialState);
+  const stateRef = useRef(state);
+  const wrappedSetState = useCallback((value: T) => {
+    stateRef.current = value;
+    setState(value);
+  }, []);
+  const getState = useCallback(() => stateRef.current, []);
+  return [state, wrappedSetState, getState];
+}
```

Comment:
I know you just moved this as part of this change, but this hook is really interesting to me. It's like, giving you all the instantly-updating behavior of a ref but still ensuring state updates happen normally. Neat!